plumbing/api.go

```diff
@@ -118,41 +128,23 @@ func (api *API) ConfigGet(dottedPath string) (interface{}, error) {
 
 // ChainHead returns the head tipset
 func (api *API) ChainHead(ctx context.Context) types.TipSet {
-	return api.chain.Head()
-}
-
-// GetRecentAncestorsOfHeaviestChain returns the recent ancestors of the
-// `TipSet` with height `descendantBlockHeight` in the heaviest chain.
-func (api *API) GetRecentAncestorsOfHeaviestChain(ctx context.Context, descendantBlockHeight *types.BlockHeight) ([]types.TipSet, error) {
-	return chain.GetRecentAncestorsOfHeaviestChain(ctx, api.chain, descendantBlockHeight)
+	return api.chain.Head(ctx)
 }
 
 // ChainLs returns a channel of tipsets from head to genesis
-func (api *API) ChainLs(ctx context.Context) <-chan interface{} {
-	return api.chain.BlockHistory(ctx, api.chain.Head())
+func (api *API) ChainLs(ctx context.Context) <-chan *chn.ChainLsResult {
+	return api.chain.Ls(ctx)
 }
 
-// ActorGet returns an actor from the latest state on the chain
-func (api *API) ActorGet(ctx context.Context, addr address.Address) (*actor.Actor, error) {
-	state, err := api.chain.LatestState(ctx)
-	if err != nil {
-		return nil, err
-	}
-	return state.GetActor(ctx, addr)
-}
-
-// ActorLs returns a slice of actors from the latest state on the chain
-func (api *API) ActorLs(ctx context.Context) (<-chan state.GetAllActorsResult, error) {
-	st, err := api.chain.LatestState(ctx)
-	if err != nil {
-		return nil, err
-	}
-	return state.GetAllActors(ctx, st), nil
+// ChainGetBlock gets a block by CID
+func (api *API) ChainGetBlock(ctx context.Context, id cid.Cid) (*types.Block, error) {
+	return api.chain.GetBlock(ctx, id)
 }
 
-// BlockGet gets a block by CID
-func (api *API) BlockGet(ctx context.Context, id cid.Cid) (*types.Block, error) {
-	return api.chain.GetBlock(ctx, id)
+// GetRecentAncestorsOfHeaviestChain returns the recent ancestors of the
+// `TipSet` with height `descendantBlockHeight` in the heaviest chain.
+func (api *API) GetRecentAncestorsOfHeaviestChain(ctx context.Context, descendantBlockHeight *types.BlockHeight) ([]types.TipSet, error) {
```

Comment:
This plumbing was only necessary because of the `SampleChainRandomness` method in porcelain. I moved said method into plumbing and removed `GetRecentAncestorsOfHeaviestChain`.